### PR TITLE
implemented the RemoveFavouriteUseCase

### PR DESCRIPTION
**Pull request by choisar4**
I've implemented the remove favourite use case.

A design conventions that I wanted some feedback on:

For now, if a user has no favourites, then they are not included in the Favourites.csv at all. I've been able to implement the data persistence for this pretty easily with the DataAccessObject/Interface and I have not gotten any issues regarding this design decision so far. But if anyone gets any errors regarding this, please let me know.

### CODE REVIEW

File: src/entity/FavouritesList.java [hunk 1]
```diff
@@ -24,11 +23,18 @@ public void remove(String restaurantID){
...
     @Override
     public String toString(){
+        if (favouritesList.isEmpty()){
+            return "";
+        }
         StringBuilder result = new StringBuilder();
         for (Restaurant favourite : favouritesList){
             result.append(favourite.getRestaurantID()).append(",");
         }
         result.delete(result.length() - 1, result.length());
```
Comment (OulinG138) [inline src/entity/FavouritesList.java:33]: You have the option to use join method to fulfill your purpose here.

File: src/view/RestaurantView.java [hunk 2]
```diff
@@ -105,6 +104,12 @@ public void propertyChange(PropertyChangeEvent evt) {
         info.add(restaurantAddress);
         info.add(restaurantPhoneNumber);
         info.add(restaurantCategories);
+
```
Comment (OulinG138) [inline src/view/RestaurantView.java:107]: If your action here involves toggling the visibility of the `addToFavorites` button, you can utilize `addToFavorites.setVisible(false)` inside `actionPerformed` to hide it, and similarly to show it again.

File: src/interface_adapter/remove_favourite/RemoveFavouritePresenter.java [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+        }
+        viewFavouritesViewModel.setState(viewFavouritesState);
+        viewFavouritesViewModel.firePropertyChanged();
+        viewManagerModel.setActiveView(viewFavouritesViewModel.getViewName());
+        viewManagerModel.firePropertyChanged();
+        RemoveFavouriteState removeFavouriteState = removeFavouriteViewModel.getState();
+        removeFavouriteState.setMessage(removeFavouriteOutputData.getMessage());
+        removeFavouriteViewModel.setState(removeFavouriteState);
+        removeFavouriteViewModel.firePropertyChanged();
+        viewManagerModel.setActiveView(removeFavouriteViewModel.getViewName());
```
Comment (OulinG138) [inline src/interface_adapter/remove_favourite/RemoveFavouritePresenter.java:37]: Why setting active view to two distinct views in one method call?

File: src/view/RestaurantView.java [hunk 4]
```diff
@@ -105,6 +104,12 @@ public void propertyChange(PropertyChangeEvent evt) {
...
+        if (state.getPreviousView().equals("view restaurants")) {
+            buttons.add(addToFavourite);
+        } else if (state.getPreviousView().equals("view favourites")) {
+            buttons.remove(addToFavourite);
+        }
```
Comment (OulinG138) [inline src/view/RestaurantView.java:112]: If your action here involves toggling the visibility of the `addToFavorites` button, you can utilize `addToFavorites.setVisible(false)` inside `actionPerformed` to hide it, and similarly to show it again.

File: src/interface_adapter/remove_favourite/RemoveFavouritePresenter.java [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+        viewManagerModel.firePropertyChanged();
```
Comment (OulinG138) [inline src/interface_adapter/remove_favourite/RemoveFavouritePresenter.java:38]: Why setting active view to two distinct views in one method on line 31 and 37

File: src/entity/FavouritesList.java [hunk 6]
```diff
@@ -24,11 +23,18 @@ public void remove(String restaurantID){
...
         return result.toString();
     }
```
Comment (OulinG138) [inline src/entity/FavouritesList.java:35]: Using join method can help achieve your goal, making your intentions clearer to the readers in this context.
```java
List<String> restaurantIds = new ArrayList<>();
for (Restaurant favourite : favouritesList) {
    restaurantIds.add(favourite.getRestaurantID());
}
String result = String.join(",", restaurantIds);
```

File: src/view/RestaurantView.java (hunk 4 above)
Comment (choisar4) [reply src/view/RestaurantView.java:112]: Thanks! This is a much better solution.

File: src/interface_adapter/remove_favourite/RemoveFavouritePresenter.java (hunk 5 above)
Comment (choisar4) [reply src/interface_adapter/remove_favourite/RemoveFavouritePresenter.java:38]: You're right line 31 is redundant. I'll make sure to delete it.

File: src/entity/FavouritesList.java (hunk 6 above)
Comment (choisar4) [reply src/entity/FavouritesList.java:36]: Good Idea. I don't actually use this method right now, but I kept it in if we ever want to improve the UI of the View Favourites View.

**Comment by choisar4** [comment]
I made the FavouritesList an Iterable so we can can say that we implemented a Design Pattern that we learned in class.

**Review by lisazhong13 — COMMENTED** [review]
I merged the branch to my end and it works well, I will merge this into main

**Pull request merged**